department-of-veterans-affairs/va.gov-cms
# Reapply "VACMS-16853: next build branches ftw (#17154)" (#17261)

## Description

closes #16853

## Testing done

- Updated some tests for git information/API/frontend. 
- Manually tested locally by running script.
- Tested on Tugboat

## QA steps

1. Log in and go to "/admin/content/deploy/next"
   - [x] Confirm you can select versions of next-build and vets-website from git references
   - [x] Confirm there is no build in progress
   - [x] No request or lock files should exist
   - [x] The versions for next-build and vets-website should be "__default"
2. Submit the form with two branches chosen for next-build and vets-website
   - [x] Confirm that a build request file exists. 
   - [x] No build should have started yet so no lock file should exist.
   - [x] The versions for next-build and vets-website should be reported as what you selected.
2. Refresh the page a few times until the background process kicks off from reading the build request file.
   - [x] Confirm that the form fields are locked during the build.
   - [x] Confirm you can view the build log file and it is adding data until the build finishes.
4. Refresh the form page after the build is complete
   - [x] Confirm the disabled fields are no longer disabled.
   - [x] View the log to see that the build finished. It might have an error, but that will be addressed in future tickets.
 5. Navigate to "/admin/content/deploy"
    - [x] Confirm that the vets-website version is not the same as the one you chose on "/admin/content/deploy/next".

### Definition of Done


### Select Team for PR review
